straxen/plugins/nveto_pulse_processing.py

```diff
@@ -0,0 +1,804 @@
+
+    Returns:
+        np.array: Array of the nveto_pulses_dtype.
+    """
+    # TODO: None of the width estimates is robust against bipolar noise...
+    # TODO: Baseline part is not subtracted yet.
+    dt = records['dt'][0]
+    record_offset = np.zeros(np.max(pulses['channel'])+1, np.int32)
+
+    for pind, pulse in enumerate(pulses):
+        # Frequently used quantaties:
+        ch = pulse['channel']
+        t = pulse['time']
+
+        # Getting data and baseline of the event:
+        ro = record_offset[ch]
+        data, ro = get_pulse_data(records, pulse, start_index=ro)
+        record_offset[ch] = ro
+
+        area, pos_deciles, height, amp_ind = _compute_area_deciles_amplitude(data)
+        amp_time = int(amp_ind * dt)
+
+        # Computing FWHM:
+        left_edge, right_edge = get_fwxm(data, amp_ind, 0.5)
+        left_edge = left_edge * dt + dt / 2
+        right_edge = right_edge * dt - dt / 2
+        width = right_edge - left_edge
+
+        # Computing FWTM:
+        left_edge_low, right_edge = get_fwxm(data, amp_ind, 0.1)
+        left_edge_low = left_edge_low * dt + dt / 2
+        right_edge = right_edge * dt - dt / 2
+        width_low = right_edge - left_edge_low
+
+        res = result_buffer[pind]
+        res['time'] = t
+        res['endtime'] = pulse['endtime']
+        res['channel'] = ch
+        res['area'] = area / to_pe[ch]
+        res['height'] = height * volts_per_adc
+        res['amp_time'] = amp_time
+        res['width'] = width
+        res['left'] = left_edge
+        res['low_left'] = left_edge_low
+        res['low_width'] = width_low
+        res['area_decile'][1:] = (pos_deciles[11:] - pos_deciles[:10][::-1]) * dt
+        res['area_midpoint'] = pos_deciles[10] * dt
+        res['area_decile_from_midpoint'][:] = (pos_deciles[::2] - pos_deciles[10]) * dt
+        res['split_i'] = pulse['split_i']
+    return result_buffer
+
+@numba.njit(cache=True, nogil=True)
+def _compute_area_deciles_amplitude(data):
+    """
+    Function which computes the area, area_decile_fractions, amplitude
+    and the position of the amplitude for a given waveform.
+
+    Args:
+        data (np.array): Waveform data
+
+    Returns:
+        float: area
+        np.arary: area_decile_fractions
+        float: amplitude
+        int: position of the amplitude
+    """
+
+    total_area = np.sum(data)
+    amp = 0
+    amp_ind = 0
+    area = 0
+    needed_decile = 1
+    prev_total_fraction = 0
+    current_total_fraction = 0
+    compute_deciles = True
+    area_decils = np.arange(0, 1.01, 0.05)
+    pos_deciles = np.zeros(len(area_decils), dtype=np.float32)
+
+    for ind, d in enumerate(data):
+        area += d
+        # Getting max amplitude:
+        if d > amp:
+            amp = d
+            amp_ind = ind
+
+        #  Check if we exceeded area fraction:
+        # -------------------------
+        # Area decile calculation. Code adapted
+        # from strax.processing.peak_properties_index_of_fraction
```

Comment:
Same here: could we use a common code instead of copy-paste-modify?

Reply:
I agree here as well. I think the problem was, that the original code requires an array of pulses which would mean in our case another time looping over all events. I will check which modification will be necessary.

Reply:
Hej I would like to make the following proposal for a change in strax:

https://gist.github.com/WenzDaniel/55a2720ad836b43f64189a64704d8f99

If it is okay with you I will make the required changes in strax and in my plugin.

Reply:
Okay I started now a https://github.com/AxFoundation/strax/pull/257 for the required changes in strax.